lib/rustler_precompiled.ex

```diff
@@ -802,6 +813,42 @@ defmodule RustlerPrecompiled do
     end)
   end
 
+  defp retry_attempts(options) do
+    retry? = Keyword.get(options, :retry, true)
+
+    if retry? do
+      value = Keyword.get(options, :retry_attempts, 3)
+
+      if not is_integer(value) or value < 1 or value > 15,
+        do: raise("attempts should be between 1 and 15. Got: #{inspect(value)}")
+
+      value
+    else
+      1
+    end
+  end
+
+  defp with_retry(fun, attempts) do
+    task = Task.async(fun)
+
+    Enum.reduce_while(1..attempts, Task.await(task), fn count, partial_result ->
+      case partial_result do
+        {:ok, _} ->
+          {:halt, partial_result}
+
+        err ->
+          Logger.warn("Attempt #{count} failed with #{inspect(err)}")
+
+          wait_in_ms = :rand.uniform(count * 2_000)
+          Process.sleep(wait_in_ms)
+
+          task = Task.async(fun)
+
+          {:cont, Task.await(task)}
```

Comment:
This has a default timeout of 5_000. Is this intentional?

Reply:
I honestly didn't think about that. Do you think it's better to increase this? Use `:infinity`? I think it can be hard to tell an exact number because the artifacts can vary in size, and the network can vary in speed. So maybe `:infinity` is a good idea here. WDYT?

Reply:
infinity is fine, yes.